### PR TITLE
Update components documentation

### PR DESCRIPTION
**Pull request by fiver-watson**
Overhaul the Components page in the user manual to describe more components, add a diagram, and generalize things a bit more away from the actual technologies we use.

### CODE REVIEW

**Comment by djjuhasz** [comment]
@fiver-watson I'll wait until the CI is passing before reviewing this PR.

**Comment by fiver-watson** [comment]
Ok, I think it should be fixed now, @djjuhasz . That said, not loving the strictness of the linter so far: 

* I used a `<br />` element to add some space between the page TOC and the image, bc I dislike how everything gets crammed together. My understanding was that HTML could always be used with markdown, since that is how it is rendered on the web anyway. But the linter made me remove it. is there some other way to add space between elements that you know of? This is most regularly an issue after bulleted lists, I find (it's the same on mediawiki sites)
* It also complained about more than 1 linebreaks between elements, even though this does not affect how the page renders at all. This was for my own legibility between sections. Is there a reason why such a thing is bad / why such a check is  needed?

In any case, I fixed it. But if you have any alternative suggestions for how to achieve was I was trying to do, plz let me know!

**Comment by djjuhasz** [comment]
@fiver-watson it looks like we are using [markdownlint](https://github.com/DavidAnson/markdownlint/) to lint the Enduro markdown.  The README.md for markdownlint describes the general rationale for the strictness of the linting (for consistency) and the individual rules usually have a "rationale" section about the rule.  At the end of the day I think most linting decisions are arbitrary and there are always rules that I don't like personally, but I think it's worthwhile to avoid wasted time and effort debating personal preferences.

Looking at the Enduro docs, I don't see any other sections like your "Jump to:" list of links, and I notice it duplicates the mkdocs navigation menu on the left of the screen:

image

I think it's better to remove the "Jump to:" links and let users use the generated index instead.

Once the "Jump to:" list is removed, I think the image would look better without the separator line above it.  That's my personal taste though, I won't argue if you feel strongly that the separator is necessary.

File: docs/src/user-manual/components.md [hunk 1]
```diff
@@ -1,30 +1,165 @@
...
+
+## Processing storage service
+
+This component acts as Enduro's storage back-end for any local package
+interactions that are needed as part of ingest and preservation workflows.
+Examples include:
+
+* Uploading Submission Information Packages (SIPs) for ingest
+* Depositing failed ingest packages after a workflow error
+* Passing Processing Information Packages (PIPs) to the [preservation
```
Comment (djjuhasz) [inline docs/src/user-manual/components.md]: We don't use MinIO to pass PIPs to a3m or AM.

File: docs/src/user-manual/components.md [hunk 2]
```diff
@@ -1,30 +1,165 @@
...
+Enduro  uses the Go programming language ([GoLang](https://go.dev/)) for its
+application code.
+
+## Application Programming Interface (API)
+
+The [API](https://en.wikipedia.org/wiki/API) is how Enduro Ingest communicates
+with its [user interface](#user-interface), and how external users and
+applications can interact with Enduro. The Enduro Ingest API defines multiple
+services which map to resources in [REST](https://en.wikipedia.org/wiki/REST) or
+service declarations in [gRPC](https://grpc.io/). Services define their own
```
Comment (djjuhasz) [inline docs/src/user-manual/components.md]: The Enduro API doesn't currently have any gRPC services, it is all HTTP/REST at this point.

File: docs/src/user-manual/components.md [hunk 3]
```diff
@@ -1,30 +1,165 @@
...
+
+Enduro's web-based user interface is the graphical method by which operators can
+interact with Enduro to see and manage ingest and preservation workflows, view
+information about packages, and more. Key information from other components and
+applications (such as tasks executed by the [preservation
+engine](#preservation-engine) in a workflow) is also collected via API and shown
+in the Enduro Ingest user interface when relevant.
+
+Enduro's user interface is built using [Vue.js](https://vuejs.org/) and
+[Bootstrap](https://getbootstrap.com/)
```
Comment (djjuhasz) [inline docs/src/user-manual/components.md]: Missing period at the end of the sentence.

File: docs/src/user-manual/components.md [hunk 4]
```diff
@@ -1,30 +1,165 @@
...
+with its [user interface](#user-interface), and how external users and
+applications can interact with Enduro. The Enduro Ingest API defines multiple
+services which map to resources in [REST](https://en.wikipedia.org/wiki/REST) or
+service declarations in [gRPC](https://grpc.io/). Services define their own
+methods, errors, and so on.
+
+Enduro's application code uses the [Goa design language](https://goa.design/) to
+structure its API design. The API itself uses [Swagger](https://swagger.io/) to
+generate its documentation and ensure it conforms to
+[OpenAPI](https://www.openapis.org/) design specifications
```
Comment (djjuhasz) [inline docs/src/user-manual/components.md]: Missing a final period.

File: docs/src/user-manual/components.md [hunk 5]
```diff
@@ -1,30 +1,165 @@
...
+engine](#preservation-engine) in a workflow) is also collected via API and shown
+in the Enduro Ingest user interface when relevant.
+
+Enduro's user interface is built using [Vue.js](https://vuejs.org/) and
+[Bootstrap](https://getbootstrap.com/)
+
+## Preservation engine
+
+This component manages the transformation of Processing Information Packages
+(PIPs) into Archival Information Packages (AIPs) in a preservation workflow
```
Comment (djjuhasz) [inline docs/src/user-manual/components.md]: I think you can just cut "in a preservation workflow during processing" from this sentence.

File: docs/src/user-manual/components.md [hunk 6]
```diff
@@ -1,30 +1,165 @@
...
+  Archivematica that is wholly focused on AIP creation. It does not have
+  external dependencies, integration with access systems, search capabilities,
+  or a graphical interface. It was designed to reduce the bulk of
+  Archivematica's extraneous functions for users operating at a large scale who
+  are more focused on throughput of digital objects for preservation.
+
+## Preservation storage service
+
+This component acts as an interface between the
+[preservation engine](#preservation-engine) and whatever storage has been
```
Comment (djjuhasz) [inline docs/src/user-manual/components.md]: "whatever storage" -> "storage devices"?

File: docs/src/user-manual/components.md [hunk 7]
```diff
@@ -1,30 +1,165 @@
...
+configured for the long-term preservation of AIPs produced by Enduro workflows.
+The storage service manages the storage, movement, replication, and deletion of
+packages, and other related functions such as location management, encryption,
+fixity checking, and more.
+
+Currently Enduro uses the [Archivematica](https://archivematica.org) Storage
+Service ([AMSS](https://github.com/artefactual/archivematica-storage-service))
+when using Archivematica as the system's preservation engine. When using a3m,
+Enduro uses its own lightweight storage functionality to cover basic storage
+servicre requirements.
```
Comment (djjuhasz) [inline docs/src/user-manual/components.md]: typo: "servicre"

File: docs/src/user-manual/components.md [hunk 8]
```diff
@@ -1,30 +1,165 @@
...
+
+-----
+
+* [Back to top](#components)
```
Comment (djjuhasz) [inline docs/src/user-manual/components.md]: I think you can cut these links too — the user can navigate to other pages or the top of this page with auto-generated TOC.

**Review by djjuhasz — CHANGES_REQUESTED** [review]
Looks really good overall @fiver-watson, good job summarizing the technical details. :)  I've suggested a handful of minor changes.

File: docs/src/user-manual/components.md (hunk 1 above)
Comment (fiver-watson) [reply docs/src/user-manual/components.md]: Got it, thanks. Should be fixed.

File: docs/src/user-manual/components.md (hunk 2 above)
Comment (fiver-watson) [reply docs/src/user-manual/components.md]: Ah good to know - i literally copied that from a comment in the code. Removed!

File: docs/src/user-manual/components.md (hunk 3 above)
Comment (fiver-watson) [reply docs/src/user-manual/components.md]: .

File: docs/src/user-manual/components.md (hunk 4 above)
Comment (fiver-watson) [reply docs/src/user-manual/components.md]: .

File: docs/src/user-manual/components.md (hunk 5 above)
Comment (fiver-watson) [reply docs/src/user-manual/components.md]: Good suggestion, done.

File: docs/src/user-manual/components.md (hunk 6 above)
Comment (fiver-watson) [reply docs/src/user-manual/components.md]: Done, thanks for the suggestion

File: docs/src/user-manual/components.md (hunk 7 above)
Comment (fiver-watson) [reply docs/src/user-manual/components.md]: :upside_down_face:

File: docs/src/user-manual/components.md (hunk 8 above)
Comment (fiver-watson) [reply docs/src/user-manual/components.md]: Sure

**Comment by fiver-watson** [comment]
Thanks for the feedback @djjuhasz ! I think I have addressed it all. I will wait to see if @sallain has any further suggestions (and that my changes look good on second review).

**Review by sallain — APPROVED** [review]

File: docs/src/user-manual/components.md [hunk 9]
```diff
@@ -1,30 +1,141 @@
...
-
-[a3m](https://github.com/artefactual-labs/a3m) is a streamlined version of
-[Archivematica](https://archivematica.org) that is wholly focused on AIP
-creation. It does not have external dependencies, integration with access
-systems, search capabilities, or a graphical interface. It was designed to
-reduce the bulk of Archivematica's extraneous functions for users operating at a
-large scale who are more focused on throughput of digital objects for
-preservation.
+Enduro Ingest is composed of a number of different components used to provide
+its ingest and preservation functionality.
```
Comment (djjuhasz) [inline docs/src/user-manual/components.md]: I would cut "its" from this sentence.

File: docs/src/user-manual/components.md [hunk 10]
```diff
@@ -1,30 +1,141 @@
...
+
+![enduro components diagram](screenshots/enduro-ingest-components.jpg)
+
+## Processing storage service
+
+This component acts as Enduro's storage back-end for any local package
```
Comment (djjuhasz) [inline docs/src/user-manual/components.md]: I think it's better to use the full component name to start this paragraph, e.g. "The processing storage service acts as Enduro's..."

File: docs/src/user-manual/components.md [hunk 11]
```diff
@@ -1,30 +1,141 @@
...
+
+Currently, Enduro uses [MinIO](https://min.io/) as its processing storage
+service. MinIO is a flexible, high performance object storage platform. Material
+intended for preservation can be uploaded to MinIO either through its user
+interface or via command line using the [MinIO
+client](https://min.io/docs/minio/linux/reference/minio-mc.html).
+
+## Messaging queue
+
+The Messaging queue acts as a watcher for SIPs deposited for ingest into the
```
Comment (djjuhasz) [inline docs/src/user-manual/components.md]: Either "The Messaging Queue" (as a proper name) or "The messaging queue" (as a general component type).

File: docs/src/user-manual/components.md [hunk 12]
```diff
@@ -1,30 +1,141 @@
...
+[processing storage service](#processing-storage-service). Thanks to this
+watcher, any time new content is uploaded to a designated bucket in MinIO, an
+ingest workflow is started in Enduro.
+
+Enduro ingest uses [Redis](https://redis.io/) as this messaging queue, to ensure
```
Comment (djjuhasz) [inline docs/src/user-manual/components.md]: "Enduro Ingest" as it is a proper noun.

Comment (djjuhasz) [reply docs/src/user-manual/components.md]: "...as *a* messaging queue..." (without emphasis).

File: docs/src/user-manual/components.md [hunk 13]
```diff
@@ -1,30 +1,141 @@
...
+reliable communication between the ingest application and the processing storage
+service.
+
+## Workflow engine
+
+The Workflow engine is responsible for orchestrating Enduro's workflows - that
```
Comment (djjuhasz) [inline docs/src/user-manual/components.md]: The combination of the hyphen and comma in the phrase "...Enduro's workflows, - that is, ..." looks weird to me.  Maybe just cut "that is," altogether?

Comment (djjuhasz) [reply docs/src/user-manual/components.md]: Also standardize the capitalization of "The Workflow Engine" or "The workflow engine".

File: docs/src/user-manual/components.md [hunk 14]
```diff
@@ -1,30 +1,141 @@
...
+resulting in a reliable platform that can process digital objects for
+preservation in a highly automated environment.
+
+Currently, Enduro Ingest uses [Temporal](https://temporal.io/) as its workflow
+engine.
+
+## Application code
+
+The application code is where the business logic guiding Enduro's workflows is
+written. It is the glue that helps each component in the Enduro Ingest stack act
```
Comment (djjuhasz) [inline docs/src/user-manual/components.md:53]: "the Enduro Ingest stack act" is a bit awkward, maybe "The code is the glue that helps each component of Enduro Ingest act in concert,..."?

File: docs/src/user-manual/components.md [hunk 15]
```diff
@@ -1,30 +1,141 @@
...
+in concert, to assist operators achieve their preservation goals.
+
+Enduro  uses the Go programming language ([GoLang](https://go.dev/)) for its
```
Comment (djjuhasz) [inline docs/src/user-manual/components.md:56]: Extra space after "Enduro".

Comment (djjuhasz) [reply docs/src/user-manual/components.md:56]: I think it's better to add the Go hyperlink directly to the sentence text instead of as a separate word, e.g. "the [Go programming language](https://go.dev/) for its..."

File: docs/src/user-manual/components.md [hunk 16]
```diff
@@ -1,30 +1,141 @@
...
+## Application Programming Interface (API)
+
+The [API](https://en.wikipedia.org/wiki/API) is how Enduro Ingest communicates
+with its [user interface](#user-interface), and how external users and
+applications can interact with Enduro. The Enduro Ingest API defines multiple
+services which map to resources in [REST](https://en.wikipedia.org/wiki/REST).
+Services define their own methods, errors, and so on.
+
+Enduro's application code uses the [Goa design language](https://goa.design/) to
+structure its API design. The API itself uses [Swagger](https://swagger.io/) to
```
Comment (djjuhasz) [inline docs/src/user-manual/components.md:68]: I think Goa generates the OpenAPI documentation itself, without depending on any Swagger (Smarbear) generation tools.  I may be wrong about this though, @jraddaoui can you confirm or deny?

File: docs/src/user-manual/components.md [hunk 17]
```diff
@@ -1,30 +1,141 @@
...
+generate its documentation and ensure it conforms to
+[OpenAPI](https://www.openapis.org/) design specifications.
+
+## Identity & access management (IAM)
+
+This component provides authentication as well as role-based access control
```
Comment (djjuhasz) [inline docs/src/user-manual/components.md]: I think it's better to start the paragraph with the component name, e.g. "The Identity and Access Management (IAM) component provides authentication..." and remove the acronym from the title.

File: docs/src/user-manual/components.md [hunk 18]
```diff
@@ -1,30 +1,141 @@
...
+OpenID Connect, OAuth 2.0, and SAML.
+
+## User interface
+
+Enduro's web-based user interface is the graphical method by which operators can
+interact with Enduro to see and manage ingest and preservation workflows, view
+information about packages, and more. Key information from other components and
+applications (such as tasks executed by the [preservation
+engine](#preservation-engine) in a workflow) is also collected via API and shown
+in the Enduro Ingest user interface when relevant.
```
Comment (djjuhasz) [inline docs/src/user-manual/components.md:94]: You used "Enduro" instead of "Enduro Ingest" to start this section — I think it would be better to standardize on one or the other to avoid confusion about whether they are two different applications.

File: docs/src/user-manual/components.md [hunk 19]
```diff
@@ -1,30 +1,141 @@
...
+
+Enduro's user interface is built using [Vue.js](https://vuejs.org/) and
+[Bootstrap](https://getbootstrap.com/).
+
+## Preservation engine
+
+This component manages the transformation of Processing Information Packages
```
Comment (djjuhasz) [inline docs/src/user-manual/components.md]: Again, I'd prefer the component name to start the section rather than "This component...".

File: docs/src/user-manual/components.md [hunk 20]
```diff
@@ -1,30 +1,141 @@
...
+  functional model and other digital preservation standards and best practices,
+  define preservation policies, manage processing configurations, and more.
+  Archivematica includes a web-based dashboard for user interaction, as well as
+  a set of [API](https://en.wikipedia.org/wiki/API) endpoints that Enduro uses
+  to integrate with Archivematica and display preservation task outcomes in the
+  Enduro user interface.
+* [a3m](https://github.com/artefactual-labs/a3m) is a streamlined version of
+  Archivematica that is wholly focused on AIP creation. It does not have
+  external dependencies, integration with access systems, search capabilities,
+  or a graphical interface. It was designed to reduce the bulk of
```
Comment (djjuhasz) [inline docs/src/user-manual/components.md]: "Reduce the bulk of Archivematica's extraneous functions" seems a bit wordy and I think it would be good to clarify that the removed features are only extraneous within the context of bulk AIP creation. Maybe "A3m was designed to remove many Archivematica features that are not required for operating at a large scale to improve AIP creation throughput.", or something similar?

File: docs/src/user-manual/components.md [hunk 21]
```diff
@@ -1,30 +1,141 @@
...
+  Archivematica's extraneous functions for users operating at a large scale who
+  are more focused on throughput of digital objects for preservation.
+
+## Preservation storage service
+
+This component acts as an interface between the [preservation
```
Comment (djjuhasz) [inline docs/src/user-manual/components.md]: "The preservation storage service acts as an interface..."

**Review by djjuhasz — CHANGES_REQUESTED** [review]
@fiver-watson I have made a lot of little suggestions, but none of them are deal breakers so feel free to ignore any of them that you disagree with.

File: docs/src/user-manual/components.md (hunk 9 above)
Comment (fiver-watson) [reply docs/src/user-manual/components.md]: :scissors:

File: docs/src/user-manual/components.md (hunk 10 above)
Comment (fiver-watson) [reply docs/src/user-manual/components.md]: :+1:

File: docs/src/user-manual/components.md (hunk 12 above)
Comment (fiver-watson) [reply docs/src/user-manual/components.md]: fixed

File: docs/src/user-manual/components.md (hunk 13 above)
Comment (fiver-watson) [reply docs/src/user-manual/components.md]: :scissors:

File: docs/src/user-manual/components.md (hunk 18 above)
Comment (fiver-watson) [reply docs/src/user-manual/components.md:94]: Will do so. In my head I was trying to consider what might be common across future Enduro components  / services vs exclusive to Enduro Ingest - like my hope is that any other team that starts building another Enduro service will use Keycloak for any IAM needs.... but yah, too soon for that kind of thinking anyway. I will just make this say Enduro Ingest for now.

Comment (fiver-watson) [reply docs/src/user-manual/components.md:94]: Although, in reading it through, there are definitely places where that is awkward, and I also don't think it's that confusing. It's like calling it the Storage Service instead of the Archivematica Storage Service every time, or like using both Access to Memory and AtoM. I think people can figure out what we mean in context, especially considering that right now there's not really any other public-facing Enduro components.... 

But, point taken in this case!

File: docs/src/user-manual/components.md (hunk 19 above)
Comment (fiver-watson) [reply docs/src/user-manual/components.md]: :+1:

File: docs/src/user-manual/components.md (hunk 20 above)
Comment (fiver-watson) [reply docs/src/user-manual/components.md]: sure, works for me.

**Comment by fiver-watson** [comment]
ok @djjuhasz I have implemented feedback fixes

**Review by djjuhasz — APPROVED** [review]
Looks good! Thanks @fiver-watson. :+1:
